src/lib/types.ts

```diff
@@ -9,3 +9,263 @@ export type AnyJson =
+} | {
+	TYPE: T;
+	VALUE: V[];
+	ACCESS: OSCAccess;
+	CLIPMODE: OSCClipMode;
+	EXTENDED_TYPE: "list"
+});
+
+export type OSCQueryListValue<T extends string = string, V extends Array<OSCValue> = Array<OSCValue>> = OSCQueryBaseNode & {
+	TYPE: T;
+	VALUE: V;
+	ACCESS: OSCAccess;
+	CLIPMODE: OSCClipMode;
+	EXTENDED_TYPE: "list"
+};
+
+export type OSCQueryFalseValue = OSCQuerySingleValue<OSCQueryValueType.False, null>;
+export type OSCQueryCharValue = OSCQuerySingleValue<OSCQueryValueType.Char, string>;
+export type OSCQueryDoubleValue = OSCQuerySingleValue<OSCQueryValueType.Double64, number>;
+export type OSCQueryFloatValue = OSCQuerySingleValue<OSCQueryValueType.Float32, number>;
+export type OSCQueryInfValue = OSCQuerySingleValue<OSCQueryValueType.Infinitum, null>;
+export type OSCQueryIntValue = OSCQuerySingleValue<OSCQueryValueType.Int32, number>;
+export type OSCQueryNilValue = OSCQuerySingleValue<OSCQueryValueType.Nil, null>;
+export type OSCQueryStringValue = OSCQuerySingleValue<OSCQueryValueType.String, string>;
+export type OSCQueryTimetagValue = OSCQuerySingleValue<OSCQueryValueType.Timetag, string>;
+export type OSCQueryTrueValue = OSCQuerySingleValue<OSCQueryValueType.True, null>;
+
+export type OSCQueryUnknownValue = OSCQuerySingleValue<OSCQueryValueType.Unknown, "">;
+
+export type OSCQueryValue = OSCQueryFalseValue | OSCQueryCharValue | OSCQueryDoubleValue | OSCQueryFloatValue | OSCQueryInfValue |
+OSCQueryIntValue | OSCQueryNilValue | OSCQueryStringValue | OSCQueryTimetagValue | OSCQueryTrueValue | OSCQueryUnknownValue | OSCQueryListValue;
+
+export type OSCQueryValueRange = {
+	RANGE?: {
+		0: {
+			MIN?: number;
+			MAX?: number;
+			VALS?: Array<number>;
+		};
+	};
+}
+
+export type OSCQueryRNBOInfoState = OSCQueryBaseNode & {
+	CONTENTS: {
+		compiler_id: OSCQueryStringValue;
+		compiler_version: OSCQueryStringValue;
+		disk_bytes_available: OSCQueryStringValue;
+		supported_cmds: OSCQueryListValue;
+		system_id: OSCQueryListValue;
+		system_name: OSCQueryListValue;
+		system_processor: OSCQueryListValue;
+		unpported_cmds: OSCQueryListValue;
+		update: OSCQueryBaseNode & {
+			CONTENTS: {
+				supported: OSCQueryFalseValue | OSCQueryTrueValue;
+			};
+		}
+		version: OSCQueryListValue;
+	};
+};
+
+export type OSCQueryRNBOConfigState = OSCQueryBaseNode & {
+	CONTENTS: Record<string, OSCQueryValue>;
+};
+
+export type OSCQueryRNBOJackPortInfo = OSCQueryBaseNode & {
+	CONTENTS: {
+		audio: OSCQueryBaseNode & {
+			CONTENTS: {
+				sinks: OSCQueryListValue<string, string[]>;
+				sources: OSCQueryListValue<string, string[]>;
+			};
+		};
+		midi: OSCQueryBaseNode & {
+			CONTENTS: {
+				sinks: OSCQueryListValue<string, string[]>;
+				sources: OSCQueryListValue<string, string[]>;
+			};
+		};
+		aliases: OSCQueryBaseNode & {
+			CONTENTS: Record<string, OSCQueryListValue<string, string[]>>;
+		};
+	};
+};
+
+export type OSCQueryRNBOJackState = OSCQueryBaseNode & {
+	CONTENTS: {
+		active: OSCQueryFalseValue | OSCQueryTrueValue;
```

Comment:
maybe should define a type `type OSCQueryBoolValue = OSCQueryFalseValue | OSCQueryTrueValue;` and use those here?